zerver/tests/test_slack_importer.py

```diff
@@ -0,0 +1,409 @@
+
+    @mock.patch("zerver.lib.slack_data_to_zulip_data.open_data_file")
+    @mock.patch("zerver.lib.slack_data_to_zulip_data.get_model_id", return_value=1)
+    def test_users_to_zerver_userprofile(self, mock_get_model_id: mock.Mock,
+                                         mock_open_data_file: mock.Mock) -> None:
+        user_data = [{"id": "U0CBK5KAT",
+                      "is_admin": True,
+                      "is_bot": False,
+                      "is_owner": True,
+                      'name': 'Jane',
+                      "real_name": "Jane Doe",
+                      "deleted": False,
+                      "profile": {"image_32": "https:\/\/secure.gravatar.com\/avatar\/random.png",
+                                  "email": "jane@foo.com"}},
+                     {"id": "U08RGD1RD",
+                      "name": "john",
+                      "deleted": False,
+                      "real_name": "John Doe",
+                      "profile": {"image_32": "", "email": "jon@gmail.com"}},
+                     {"id": "U09TYF5Sk",
+                      "name": "Bot",
+                      "real_name": "Bot",
+                      "is_bot": True,
+                      "deleted": False,
+                      "profile": {"image_32": "https:\/\/secure.gravatar.com\/avatar\/random1.png",
+                                  "email": "bot1@zulipchat.com"}}]
+        mock_open_data_file.return_value = user_data
+        test_added_users = {'U08RGD1RD': 2,
+                            'U0CBK5KAT': 1,
+                            'U09TYF5Sk': 3}
+        slack_data_dir = './random_path'
+        timestamp = int(timezone_now().timestamp())
+        zerver_userprofile, added_users = users_to_zerver_userprofile(slack_data_dir, 1, timestamp, 'test_domain')
+
+        self.assertDictEqual(added_users, test_added_users)
+
+        self.assertEqual(zerver_userprofile[0]['id'], test_added_users['U0CBK5KAT'])
+        self.assertEqual(len(zerver_userprofile), 3)
+        self.assertEqual(zerver_userprofile[0]['id'], 1)
+        self.assertEqual(zerver_userprofile[0]['is_realm_admin'], True)
+        self.assertEqual(zerver_userprofile[0]['is_active'], True)
+        self.assertEqual(zerver_userprofile[1]['is_staff'], False)
+        self.assertEqual(zerver_userprofile[1]['is_bot'], False)
+        self.assertEqual(zerver_userprofile[1]['enable_desktop_notifications'], True)
+        self.assertEqual(zerver_userprofile[2]['bot_type'], 1)
+
+    @mock.patch("zerver.lib.slack_data_to_zulip_data.get_model_id", return_value=1)
+    def test_build_defaultstream(self, mock_get_model_id: mock.Mock) -> None:
+        realm_id = 1
+        stream_id = 1
+        default_channel_general = build_defaultstream('general', realm_id, stream_id)
+        test_default_channel = {'stream': 1, 'realm': 1, 'id': 1}
+        self.assertDictEqual(test_default_channel, default_channel_general)
+        self.assertIsNone(build_defaultstream('random', 1, 1))
```

Comment:
Is it true that not every users is subscribed to channel '#random' in Slack?

Reply:
I couldn't find a documentation for it, but here we have one for the general channel :https://get.slack.help/hc/en-us/articles/220105027-The-general-channel

Reply:
The '#general''s and '#random''s equivalent in Zulip are '#announcement' and '#general'. From https://slackhq.com/how-to-manage-your-slack-channels-38801ef651f2:
> Every Slack team starts with a #general and a #random channel, and most grow from there.

Reply:
So, should I also map `random` stream to `defaultstream`?

Reply:
Yes, based on that sentence. I have tested I can unsubscribe from '#announcement' and '#general' in Zulip. So I think this should be put in the documentation that Zulip doesn't have a concept of default channel where no one can leave, but does have list of streams where everyone is subscribed to by default when they register.

Reply:
@rht I have made the changes